### PR TITLE
Added unit tests for url-utils

### PR DESCRIPTION
**Pull request by vershwal**
no issue

This PR is the first step towards migrating urlUtils to TS. This increases the test coverage of the package. 

---

> [!NOTE]
> Preserves protocol slashes in URL deduplication and expands unit test coverage across transform-ready helpers and edge cases.
> 
> - **Fix**:
>   - Update `lib/utils/deduplicate-double-slashes.js` to preserve protocol (`http://`, `https://`) while collapsing multiple slashes in paths.
> - **Tests**:
>   - Add new suites for `html-to-transform-ready`, `lexical-to-transform-ready`, `markdown-to-transform-ready`, `mobiledoc-to-transform-ready`, `to-transform-ready`, and `deduplicate-double-slashes` covering relative/page-relative URLs, subdirectories, assetsOnly/staticImageUrlPrefix, defaults, and empty inputs.
>   - Extend `absolute-to-transform-ready` with unparseable URL handling.
>   - Extend `markdown-absolute-to-transform-ready` and `markdown-relative-to-transform-ready` with default options, protocol handling, assets-only behavior, and trailing-slash/https cases; verify AST parsing is skipped when possible.
>   - Extend `plaintext-absolute/relative/to-transform-ready` with options-as-itemPath and itemPath parameter handling.
>   - Extend `relative-to-absolute` with protocol-relative matching-domain case.
>   - Extend `replace-permalink` to handle missing/null `primary_author`.
>   - Extend `strip-subdirectory-from-path` to handle non-matching paths and invalid `rootUrl`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 69a82935656f3d8d026012bce1ad9ee8542b8db9. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/url-utils/test/unit/utils/deduplicate-double-slashes.test.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    it('should handle relative paths', function () {
+        deduplicateDoubleSlashes('/path//to//file.png')
+            .should.eql('/path/to/file.png');
+    });
+
+    it('should handle protocol slashes correctly', function () {
+        deduplicateDoubleSlashes('http://example.com//path')
+            .should.eql('http:/example.com/path');
+    });
```
[the remark this answers is not in the document]
Comment (vershwal) [reply packages/url-utils/test/unit/utils/deduplicate-double-slashes.test.js]: That's a great point, thank you! Fixed
